File: backend/src/main/java/com/codeloom/backend/BackendApplication.java
package com.codeloom.backend;

import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.context.annotation.Bean;
import org.springframework.data.jdbc.repository.config.EnableJdbcAuditing;
import tools.jackson.databind.ObjectMapper;

@SpringBootApplication
@EnableJdbcAuditing
public class BackendApplication {
  public static void main(String[] args) {
    SpringApplication.run(BackendApplication.class, args);
  }

  @Bean
  ObjectMapper objectMapper() {
    return new ObjectMapper();
  }
}

File: backend/src/main/kotlin/com/codeloom/backend/BackendApplication.kt
package com.codeloom.backend

import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.runApplication
import org.springframework.context.annotation.Bean
import org.springframework.data.jdbc.repository.config.EnableJdbcAuditing
import tools.jackson.databind.JsonNode
import tools.jackson.databind.ObjectMapper
import tools.jackson.module.kotlin.jacksonObjectMapper

@SpringBootApplication
@EnableJdbcAuditing
class BackendApplication {
    @Bean
    fun objectMapper(): ObjectMapper = jacksonObjectMapper()
}

fun main(args: Array<String>) {
    runApplication<BackendApplication>(*args)
}

inline fun <reified T> JsonNode.patchValue(
    name: String,
    objectMapper: ObjectMapper,
    current: T,
): T {
    val value = get(name) ?: return current
    if (value.isNull) return current
    return objectMapper.treeToValue(value, T::class.java)
}
